src/testlib/src/yamlToBson.cpp

```diff
@@ -0,0 +1,179 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <testlib/yamlToBson.hpp>
+
+#include <optional>
+#include <variant>
+
+#include <boost/throw_exception.hpp>
+
+#include <bsoncxx/builder/basic/array.hpp>
+#include <bsoncxx/builder/basic/document.hpp>
+#include <bsoncxx/json.hpp>
+
+namespace genny::testing {
+
+namespace {
+
+
+// Helper type cribbed from value_generators
+// The value_generators version will likely disappear soon.
+class Value {
+public:
+    explicit Value(bool value) : _value{value} {}
+    explicit Value(int32_t value) : _value{value} {}
+    explicit Value(int64_t value) : _value{value} {}
+    explicit Value(double value) : _value{value} {}
+    explicit Value(std::string value) : _value{value} {}
+    explicit Value(bsoncxx::types::b_null value) : _value{value} {}
+    void appendToBuilder(bsoncxx::builder::basic::document& doc, std::string key) {
+        std::visit(
+            [&](auto&& arg) { doc.append(bsoncxx::builder::basic::kvp(std::move(key), arg)); },
+            std::move(_value));
+    }
+    void appendToBuilder(bsoncxx::builder::basic::array& arr) {
+        std::visit([&](auto&& arg) { arr.append(arg); }, std::move(_value));
+    }
+
+    static Value parseScalar(const YAML::Node& node) {
+        if (!node.IsScalar() && !node.IsNull()) {
+            std::stringstream msg;
+            msg << "Expected scalar or null got " << node.Type();
+            BOOST_THROW_EXCEPTION(InvalidYAMLToBsonException(msg.str()));
+        }
+
+        if (node.Type() == YAML::NodeType::Null) {
+            return Value{bsoncxx::types::b_null{}};
+        }
+
+        // `YAML::Node::Tag() == "!"` means that the scalar value is quoted. In that case, we want
+        // to avoid converting any numeric string values to numbers. See
+        // https://github.com/jbeder/yaml-cpp/issues/261 for more details.
+        if (node.Tag() != "!") {
+            try {
+                return Value{node.as<int32_t>()};
+            } catch (const YAML::BadConversion& e) {
+            }
+
+            try {
+                return Value{node.as<int64_t>()};
+            } catch (const YAML::BadConversion& e) {
+            }
+
+            try {
+                return Value{node.as<double>()};
+            } catch (const YAML::BadConversion& e) {
+            }
+
+            try {
+                return Value{node.as<bool>()};
+            } catch (const YAML::BadConversion& e) {
+            }
+        }
+
+        return Value{node.as<std::string>()};
+    }
+
+private:
+    using VariantType = std::variant<bool,
+                                     int32_t,
+                                     int64_t,
+                                     double,
+                                     std::string,
+                                     bsoncxx::types::b_null,
+                                     bsoncxx::document::view_or_value,
```

Comment:
> Ditto with the `std::variant<>` type.
> https://github.com/mongodb/genny/pull/183#discussion_r266061112

I'd say just have this line `bsoncxx::document::value` and the next line `bsoncxx::array::value`.

Reply:
ohai the the `bsoncxx*` types didn't actually need to be a part of the `variant` so just removed them entirely 👍